core/esmf-aspect-model-java-generator/src/test/java/org/eclipse/esmf/aspectmodel/java/StaticMetaModelJavaGeneratorTest.java

```diff
@@ -498,6 +501,34 @@ void testCharacteristicInstantiationForQuantifiableWithoutUnit() throws IOExcept
             ImmutableMap.<String, String> builder().put( "TEST_PROPERTY", expectedTestPropertyCharacteristicConstructorCall ).build(), 1 );
    }
 
+   @Test
+   void testCharacteristicInstantiationForPropertyWithExampleValue() throws IOException {
+      final TestAspect aspect = TestAspect.ASPECT_WITH_COLLECTION;
+      final StaticClassGenerationResult result = TestContext.generateStaticAspectCode().apply( getGenerators( aspect ) );
+
+      result.assertNumberOfFiles( 2 );
+
+      Map<String, Set<String>> expectedBaseAttributes = new HashMap<>();
+      expectedBaseAttributes.put( "TEST_PROPERTY", Set.of(
+            "withUrn(AspectModelUrn.fromUrn(NAMESPACE + \"testProperty\"))",
+            "withPreferredName(Locale.forLanguageTag(\"en\"), \"Test Property\")",
+            "withDescription(Locale.forLanguageTag(\"en\"), \"This is a test property.\")",
+            "withSee(\"http://example.com/\")",
+            "withSee(\"http://example.com/me\")"
+      ) );
+
+      result.assertMetaModelBaseAttributesForProperties( "MetaAspectWithCollection", expectedBaseAttributes );
+
+      final String expectedTestPropertyCharacteristicConstructorCall =
+            "new DefaultCollection(MetaModelBaseAttributes.builder().withUrn(AspectModelUrn.fromUrn(NAMESPACE + \"TestCollection\"))"
+                  + ".withPreferredName(Locale.forLanguageTag(\"en\"), \"Test Collection\").withDescription(Locale.forLanguageTag(\"en\")"
+                  + ", \"This is a test collection.\").withSee(\"http://example.com/\").build(), Optional.of(new DefaultScalar"
+                  + "(\"http://www.w3.org/2001/XMLSchema#string\")), Optional.empty())";
+
+      result.assertConstructorArgumentForProperties( "MetaAspectWithCollection",
+            ImmutableMap.<String, String> builder().put( "TEST_PROPERTY", expectedTestPropertyCharacteristicConstructorCall ).build(), 1 );
```

Comment:
I'm missing the assertion that looks for the presence of `Optional.of(new DefaultScalarValue("Example Value", new DefaultScalar("http://www.w3.org/2001/XMLSchema#string")))`, i.e., the thing we actually want to assert here.
`assertConstructorArgumentsForProperties` can not be used as-is for this, because it only checks the constructor of the Characteristic and it's not straightforward because the example value declaration is nested deep in the field declaration.

You could do something like this:

Add those methods in `StaticClassGenerationResult`:
```java
   private Stream<Node> getAllChildNodes( final FieldDeclaration field ) {
      return field.getChildNodes().stream().flatMap( this::nodeAndItsChildren );
   }

   private Stream<Node> nodeAndItsChildren( final Node node ) {
      return Stream.concat( Stream.of( node ), node.getChildNodes().stream().flatMap( this::nodeAndItsChildren ) );
   }
```

and then in `assertConstructorArgumentForProperties`, add an (optional) argument for the field declaration to check (which you use when calling the method from the test, passing the example value declaration), and use `getAllChildNodes(field).map(Node::toString)` if any of the nodes contains the declaration.